tests/test_test_tools/test_test_cases.py

```diff
@@ -151,6 +157,42 @@ def test_agent_get_exception(self):
             self.run_cli_command("config", "get", "agent.bad_key", cwd=self._get_cwd())
 
 
+class TestConfigCasesAEV(AEATestCaseEmpty):
+    """Test configuration overwrite via CLI tool"""
+
+    def test_agent_set_aev(self):
+        """Test agent skill overwrite using environment variable."""
+
+        def copy_dummy_aea_and_update_aea_config_with_env_var():
+            shutil.copytree(dummy_aea.__path__[0], agent_dir)
+            aea_config_file = agent_dir / "aea-config.yaml"
+            content = aea_config_file.read_text()
+            new_content = content + f"\nis_abstract: ${{{env_var}:bool:false}}"
+            aea_config_file.write_text(new_content)
+
+        def set_config_load_agent_return_component_config():
+            self.set_config(key_name, value, aev=True)
```

Comment:
Why are we using `agent_dir`, `key_name` and `value` from the function scope rather than passing them as arguments?

Reply:
because it doesn't matter, this is a singular test